docs/upgrades.md

```diff
@@ -0,0 +1,196 @@
+packages cannot be newer than the server they are interacting with.
+
+To see which pip packages a given server version supports, check the
+[release notes](https://github.com/sematic-ai/sematic/releases).
+
+Pip packages and the bazel integration should always be kept in sync.
+
+### Upgrading the pip package
+
+If you have installed via command line, you can do the following
+to upgrade to the latest:
+
+```bash
+pip install sematic --upgrade
+```
+
+or to install a specific version:
+
+```bash
+pip install sematic==X.Y.Z
+```
+
+If you use a `requirements.txt` file or similar, you can increase the version
+there and then install from the `requirements.txt` file.
+
+### Upgrading the bazel integration
+
+Your bazel `WORKSPACE` should contain something like the following:
+
+```python
+git_repository(
+    name = "rules_sematic",
+    remote = "https://github.com/sematic-ai/sematic.git",
+    strip_prefix = "bazel",
+    tag = "vX.Y.Z",
+)
+```
+
+To upgrade, change the `tag` to the version you wish to install (should match
+the pip package).
+
+### Upgrading the Server
+
+The server should be installed via `helm`. Make any adjustments to your `values.yaml`
+required by the latest version (See the section
+["From vX1.X2.X3 to vY1.Y2.Y3"](#from-vx1x2x3-to-vy1y2y3) below). Then:
+
+```bash
+helm upgrade sematic-server -f /path/to.values.yaml
+```
+
+## Upgrading the server past supported client versions
+
+You will always want to upgrade the server before upgrading the clients. If the
+server is being upgraded to a version that still supports the version being
+used by your clients, you can often upgrade the server without any downtime for
+users. This is one benefit of keeping up-to-date with regular upgrades! If you
+want to know what client versions a given server supports, you can look at
+the [release notes](https://github.com/sematic-ai/sematic/releases).
+
+Often, you will need to upgrade the server beyond support for the clients
+currently in use. When this happens, a recommended upgrade workflow is:
+
+1. Draft a PR in your repo that upgrades the clients
+2. Notify users that there will be an upcoming maintenance window for upgrades
+   during which they may not be able to use Sematic.
+3. When the maintenance window arrives, upgrade the server.
+4. Test the clients using the branch for your upgrade PR. Assuming all goes
+   well, merge the PR. If not, roll back the server.
+5. Notify users that the maintenance window is now complete and that they will
+   need to merge the main branch into their feature branches to begin using the
+   latest version.
+
+## From vX1.X2.X3 to vY1.Y2.Y3
+
+This section contains instructions for migrating from one
+version of Sematic to another. It does not cover every version
+delta, but only (a) from one version to a subsequent one (b)
+upgrades where there are non-trivial details to pay attention to
+when performing the upgrade.
+
+### v0.23.X to v0.24.Y
```

Comment:
This looks superfluous. Please remove.

Reply:
I think we need some way to indicate which versions the guidance applies to. What format would you prefer to present that?